src/Chapter1.hs

```diff
@@ -641,7 +657,13 @@ You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
 
-firstDigit n = error "firstDigit: Not implemented!"
+firstDigit :: Integer -> Integer
+firstDigit n
+      | isLastDigit = n
+      | otherwise   = firstDigit currentN
```

Comment:
I didn't have better name for `currentN`. What is a usual convention when dealing with something being "current" during folding?